python/plugins/processing/core/outputs.py

```diff
@@ -40,15 +41,23 @@
                        QgsProcessingOutputString,
                        QgsProcessingOutputBoolean,
                        QgsProcessingOutputFolder,
+                       QgsProcessingOutputFile,
                        QgsProcessingOutputMultipleLayers)
 
 
 def getOutputFromString(s):
     try:
-        if "|" in s and s.startswith("Output"):
+        if "|" in s and (s.startswith("QgsProcessingOutput") or s.startswith("Output")):
```

Comment:
@rldhont is there any chance you could move this logic to c++? (E.g. make QgsProcessingUtils.getOutputFromString, and just call that directly here?)

It makes me nervous seeing growth in core processing python code, when the ultimate goal is ZERO processing code in python (outside of a handful of algorithms)